packages/web/src/js/Collapse.ts

```diff
@@ -59,109 +61,149 @@ class Collapse extends BaseComponent {
       const content = this.target.children[0];
       const bounds = content.getBoundingClientRect();
-      this.target.style.height = collapsed ? `${bounds.height}px` : '0';
-      if (this.target) EventHandler.trigger(this.target, collapsed ? EVENT_SHOWN : EVENT_HIDDEN);
+      this.target.style.height = `${zeroHeight ? '0' : bounds.height}px`;
+    }
+  }
+
+  adjustCollapsibleElementHeight(open: boolean = this.state.open) {
+    this.adjustCollapsibleChildrenHeight(!open);
+    if (this.target) {
+      EventHandler.trigger(this.target, open ? EVENT_SHOWN : EVENT_HIDDEN);
     }
   }
 
-  updateTriggerElementHandler(collapsed: boolean = this.state.collapsed) {
-    const triggers = selectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
+  updateTriggerElement(open: boolean = this.state.open) {
+    const triggers = SelectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
     const updateElement = (element: Element | HTMLElement) => {
       element.setAttribute(ARIA_CONTROLS_ATTRIBUTE, this.meta.id);
-      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(collapsed));
-      element.classList.toggle(CLASSNAME_EXPANDED, collapsed);
-      if (this.meta.hideOnCollapse && collapsed) {
+      element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, String(open));
+      if (this.meta.hideOnCollapse && open) {
         element.remove();
-        this.appendNodeToParentHandler();
-        this.onDestroy();
+        this.appendNodeToParent().then(() => this.onDestroy());
       }
     };
-    this.state.collapsed = collapsed;
+    this.state.open = open;
     if (triggers.length === 1) {
       updateElement(this.element);
     } else {
-      triggers.forEach((trigger) => updateElement(trigger));
+      triggers.map((trigger) => updateElement(trigger));
     }
   }
 
-  updateCollapsibleElementHandler(collapsed: boolean = this.state.collapsed) {
+  updateCollapsibleElement(open: boolean = this.state.open) {
     if (this.target) {
-      this.target.classList.toggle(CLASSNAME_COLLAPSED, collapsed);
-      this.adjustCollapsibleElementHeightHandler(collapsed);
+      if (!open) {
+        this.adjustCollapsibleChildrenHeight();
+      }
+      this.adjustCollapsibleElementHeight(open);
+      this.target?.classList.add(CLASSNAME_TRANSITION);
+      executeAfterTransition(this.target, () => {
+        this.target?.classList.remove(CLASSNAME_TRANSITION);
+        this.target?.classList.toggle(CLASSNAME_OPEN, open);
+        if (open) {
+          this.target?.setAttribute('style', 'height: 100%');
+        } else {
+          this.target?.style.removeProperty('height');
+        }
+      });
     }
   }
 
   show() {
+    this.hideParentSiblings(this.element);
     EventHandler.trigger(this.target as HTMLElement, EVENT_SHOW);
-    this.updateTriggerElementHandler(true);
-    this.updateCollapsibleElementHandler(true);
-    EventHandler.trigger(this.target as HTMLElement, EVENT_SHOWN);
+    this.updateTriggerElement(true);
```

Comment:
The problem I see here is "why the heck is used `true`/`false` here so hardcoded?" 🤔 . I'll prefer to bind this on the existing state, eg. `this.updateTriggerElement(!this.state.open)` which works and it is clear, what the method needs as input.

Reply:
If you mean hardcoded as function parameter? For `show()` means `open = true`, so `true` is parameter. In case of `hide()` i want to close, so `open = false` and `false` is parameter.